File: Easy/177/quickSort.py
from random import randrange

def quicksort(array):
    lesser = []
    equal = []
    greater = []

    if len(array) > 1:
        pivot = array[randrange(len(array)-1)]
        for x in array:
            if x < pivot:
                lesser.append(x)
            elif x == pivot:
                equal.append(x)
            elif x > pivot:
                greater.append(x)
        return quicksort(lesser)+equal+quicksort(greater)
    else:
        return array

array = [5, 12, -3, 570, -91]
print quicksort(array)

File: Easy/177/quicksort.php
<?php

function quicksort($array){
    if( count($array) <= 1) {
        return $array;
    }

    $pivotPoint = $array[rand(0, count($array) - 1)];

    $smaller = array();
    $equal = array();
    $greater = array();
    foreach($array as $chunk) {
        if ($chunk < $pivotPoint) {
            $smaller[] = $chunk;
        } else if ($chunk === $pivotPoint) {
            $equal[] = $chunk;
        } else {
            $greater[] = $chunk;
        }
    }

    return array_merge(quicksort($smaller), $equal, quicksort($greater));
}

$array = [5, 12, -3, 570, -91];
print_r(quicksort($array));
